docs/Buttons/GetCapabilities/assets/GetCapabilities.txt

```diff
@@ -0,0 +1,62 @@
+participant App
+participant SDL
+participant HMI
+
+HMI->SDL:BC.OnReady
+SDL->HMI:Buttons.IsReady
+HMI-->SDL:Buttons.IsReady (code:0)
+SDL->SDL:  Check availability\nof `HMICapabilitiesCacheFile`
+
+opt Using cached data from `HMICapabilitiesCacheFile` - Happy Path 
+
+note over SDL:<align:center>`HMICapabilitiesCacheFile`\n**is not** available
+
+SDL->SDL:Create\n`HMICapabilitiesCacheFile`
+SDL->HMI:Buttons.GetCapabilities
+HMI-->SDL:Buttons.GetCapabilities (params)
+SDL->SDL:Cache received data\nin `HMICapabilitiesCacheFile`
+App->SDL:GetCapabilities
+SDL->SDL: Load **cached** data\nfrom **`HMICapabilitiesCacheFile`**
+SDL-->App:GetCapabilities (params)
+end
+
+opt Using default data from `hmi_capabilities.json` - No response from HMI
+
+note over SDL:<align:center>`HMICapabilitiesCacheFile`\n**is not** available
+
+SDL->SDL:  Create\n`HMICapabilitiesCacheFile`
+SDL->HMI:Buttons.GetCapabilities
+note over HMI:<align:center>No\nresponse
+App->SDL:GetCapabilities
+SDL->SDL:  Load **default** data\nfrom **`hmi_capabilities.json`**
+SDL-->App:GetCapabilities (params)
+end
+
+opt Verifying content of `HMICapabilitiesCacheFile`
+
+note over SDL:<align:center>`HMICapabilitiesCacheFile`\n**is available**
+
+SDL->HMI:Buttons.GetCapabilities
```

Comment:
@Mariia-Novosiadla please note that if `HMICapabilitiesCacheFile is available` - 
SDL will send `Buttons.GetCapabilities` request only if there are **no button capabilities in the HMICapabilitiesCacheFile**, and this applies to all capability requests.


_In case any capability is missing, SDL Core should default back to hmi_capabilities.json file for corresponding capability and should send requests to get these capabilities from HMI._
https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0249-Persisting-HMI-Capabilities-specific-to-headunit.md#functional-changes

Reply:
@Mariia-Novosiadla 
And also please add the case when `HMICapabilitiesCacheFile is available` and **buttons capabilities are present** in the HMICapabilitiesCacheFile. In this case, SDL will not send Buttons.GetCapabilities to the HMI and will use the data from the cache for usages by apps

Reply:
@LitvinenkoIra, 
Thank you for the provided comments. Please see updates in 65590c0c761ebae6acaa045d2ef442de733e4f36

> case when HMICapabilitiesCacheFile is available and buttons capabilities are present in the HMICapabilitiesCacheFile 

addressed  in:
 **Use of cached capabilities data - Happy Path**

> In case any capability is missing, SDL Core should default back to hmi_capabilities.json file for corresponding capability and should send requests to get these capabilities from HMI.

addressed in:
**Use of default capabilities data - No response from HMI** ,
 **Use of default capabilities data - Some capabilities data is missing in HMI response**

Please review and let me know if you approve these changes.